inc/Analytics/TracksAnalytics.php

```diff
@@ -0,0 +1,170 @@
+<?php declare(strict_types = 1);
+
+namespace RemoteDataBlocks\Analytics;
+
+defined( 'ABSPATH' ) || exit();
+
+use RemoteDataBlocks\Editor\BlockManagement\ConfigStore;
+
+/**
+ * Class to implement Tracks Analytics.
+ */
+class TracksAnalytics {
+	/**
+	 * The tracks instance (not using Tracks as type because it is present in MU Plugins codebase).
+	 */
+	private static object|null $instance = null;
+
+	/**
+	 * Environment configuration.
+	 */
+	private static ?EnvironmentConfig $env_config = null;
+
+	/**
+	 * Initialize Tracks Analytics based on the environment configuration.
+	 *
+	 * @param EnvironmentConfig $env_config Environment configuration.
+	 */
+	public static function init( EnvironmentConfig $env_config ): void {
+		self::$env_config = $env_config;
+
+		// Do not track on local environment.
+		if ( self::$env_config->is_local_env() ) {
+			return;
+		}
+
+		$tracks_class = self::$env_config->get_tracks_lib_class();
+		if ( ! $tracks_class ) {
+			return;
+		}
+
+		if ( self::$env_config->is_wpvip_site() || self::$env_config->is_enabled_via_filter() ) {
+			self::$instance = new $tracks_class( '', self::get_global_properties() );
+			self::setup_tracking_via_hooks();
+		}
+	}
+
+	/**
+	 * Get `Tracks` global properties to be sent with each event.
+	 */
+	public static function get_global_properties(): array {
+		$rdb_specific_props = [
+			'plugin_version' => defined( 'REMOTE_DATA_BLOCKS__PLUGIN_VERSION' ) ? constant( 'REMOTE_DATA_BLOCKS__PLUGIN_VERSION' ) : '',
+		];
+
+		return array_merge( $rdb_specific_props, self::$env_config->get_tracks_core_properties() );
+	}
+
+	private static function setup_tracking_via_hooks(): void {
+		// WordPress Dashboard Hooks.
+		add_action( 'activated_plugin', [ __CLASS__, 'track_plugin_activation' ] );
+		add_action( 'deactivated_plugin', [ __CLASS__, 'track_plugin_deactivation' ] );
+		add_action( 'save_post', [ __CLASS__, 'track_remote_data_blocks_usage' ], 10, 2 );
+	}
+
+	/**
+	 * Activation hook.
+	 *
+	 * @param string $plugin_path Path of the plugin that was activated.
+	 */
+	public static function track_plugin_activation( string $plugin_path ): void {
+		if ( ! self::$env_config->is_remote_data_blocks_plugin( $plugin_path ) ) {
+			return;
+		}
+
+		self::record_event( 'remotedatablocks_plugin_toggle', [ 'action' => 'activate' ] );
+	}
+
+	/**
+	 * Deactivation hook.
+	 *
+	 * @param string $plugin_path Path of the plugin that was deactivated.
+	 */
+	public static function track_plugin_deactivation( string $plugin_path ): void {
+		if ( ! self::$env_config->is_remote_data_blocks_plugin( $plugin_path ) ) {
+			return;
+		}
+
+		self::record_event( 'remotedatablocks_plugin_toggle', [ 'action' => 'deactivate' ] );
+	}
+
+	/**
+	 * Track usage of Remote Data Blocks.
+	 *
+	 * @param int      $post_id Post ID.
+	 * @param \WP_Post $post Post object.
+	 */
+	public static function track_remote_data_blocks_usage( int $post_id, object $post ): void {
```

Comment:
```suggestion
	public static function track_remote_data_blocks_usage( int $post_id, \WP_Post $post ): void {
```

Reply:
The `object` type is used intentionally here, as using `WP_Post` directly in tests would require a tricky approach.